uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/utils/web/PersonalizationFilter.java

```diff
@@ -0,0 +1,138 @@
+        this.personManager = personManager;
+    }
+
+    private IPersonalizer personalizer;
+
+    @Autowired
+    public void setPersonalizer(IPersonalizer personalizer) {
+        this.personalizer = personalizer;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) {}
+
+    @Override
+    public void destroy() {}
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain filterChain)
+            throws IOException, ServletException {
+        final HttpServletRequest request = (HttpServletRequest) req;
+        final HttpServletResponse response = (HttpServletResponse) resp;
+        log.debug("In PersonalizationFilter after the filterChain");
+
+        if (!enableFilter) {
+            log.debug("PersonalizationFilter is disabled - skipping.");
+            filterChain.doFilter(req, resp);
+            return;
+        }
+
+        if (!request.getMethod().equals(HttpMethod.GET.name())) {
+            log.debug(
+                    "Not a GET request - skipping the filter. Request URL: [{}] {}",
+                    request.getMethod(),
+                    request.getRequestURI());
+            filterChain.doFilter(req, resp);
+            return;
+        }
+
+        // Capture the response
+        SimpleCharacterResponseWrapper wrapper = new SimpleCharacterResponseWrapper(response);
+
+        filterChain.doFilter(req, wrapper);
+
+        PrintWriter responseWriter = response.getWriter();
+
+        if (wrapper.getContentType().contains("application/json")) {
+            final IPerson person = this.personManager.getPerson(request);
+            if (person == null) {
+                log.warn(
+                        "Person not found in Person Manager.  Not applying the personalization filter. Request URL: [{}] {}",
+                        request.getMethod(),
+                        request.getRequestURI());
+                writeToResponse(response, responseWriter, wrapper.toString());
+                return;
+            }
+
+            String originalContent = wrapper.toString();
+            if (originalContent == null) {
+                log.debug(
+                        "Original content is null. Not applying the personalization filter. Request URL: [{}] {}",
+                        request.getMethod(),
+                        request.getRequestURI());
+                return;
+            }
+
+            final String personalizedContent =
+                    personalizer.personalize(person, originalContent, request.getSession());
+            if (originalContent.equals(personalizedContent)) {
+                log.debug(
+                        "No personalizations made to the content. Request URL: [{}] {}",
+                        request.getMethod(),
+                        request.getRequestURI());
+            } else {
+                log.debug(
+                        "Personalized the content! Request URL: [{}] {}",
```

Comment:
is it a good sentence ?

Reply:
I believe so.  Doesn't seem awkward and gets the point across...